core/server/server.js

```diff
@@ -477,6 +477,17 @@ class Server {
     this.registerServices()
 
     camp.timeout = this.config.public.requestTimeoutSeconds * 1000
+    if (this.config.public.requestTimeoutSeconds > 0) {
+      camp.on('timeout', socket => {
```

Comment:
This is super-low level because the timeout event callback gives us a socket object, not a request/response.
I don't think there is a nice way to deal with this using a higher-level abstraction if we want the timeout to be applied to all requests. If we used `camp.setTimeout()` the callback param would also be a socket.